### ISSUE TITLE
How to Use  Different Optimizers with NeuralForecast Models

### ISSUE THREAD

**Issue by Amirh63** [Nixtla/neuralforecast#852]
### Description

Greetings Gentlemen,

I am currently working with the NeuralForecast library and specifically interested in training models using the Adam optimizer. I've gone through the documentation but haven't found clear guidance on how to integrate a custom optimizer, like Adam, into the training process.

Could you please provide some insights or examples on how to set up the Adam optimizer for training NeuralForecast models? I'm particularly interested in any necessary configurations or modifications needed within the library's framework to achieve this.

Additionally, if there are any best practices or recommendations for using custom optimizers with NeuralForecast, that information would be greatly appreciated.

Thank you for your time and assistance.

**Comment by candalfigomoro** [comment]
Adam is the optimizer that is already being used

**Comment by Amirh63** [comment]
How do I use a different optimizer other than Adam?

**Comment by cchallu** [comment]
Hi @Amirh63. There is currently no option to modify the optimizer using the hyperparameters. We fixed the optimizer to Adam, given that all papers proposing these models use it. But this can be a great addition, so we will add it to our future improvements. In the meantime, you can modify it manually if you clone the repo and modify the following:

In line 181 on _base_windows.py:
```
def configure_optimizers(self):
        optimizer = torch.optim.Adam(self.parameters(), lr=self.learning_rate)
        scheduler = {
            "scheduler": torch.optim.lr_scheduler.StepLR(
                optimizer=optimizer, step_size=self.lr_decay_steps, gamma=0.5
            ),
            "frequency": 1,
            "interval": "step",
        }
        return {"optimizer": optimizer, "lr_scheduler": scheduler}
```

**Comment by cchallu** [comment]
@quest-bot stash 200

**Comment by cchallu** [comment]
Solution:
- Add `optimizer` hyperparameter on all models. The default should be None, and it recieves a custom `torch.optim` object.
- If user adds an `optimizer`, pass it to the `configure_optimizers` function (see above) and use instead of default. Repeat for all base classes.
- Add a test on `core.ipynb` that when user passes a different optimizer, results are different.

**Comment by helenehanyu** [comment]
Hi @cchallu , can you make scheduler an option to?

**Comment by cchallu** [comment]
@quest-bot reward @JQGoh
